### DISCUSSION TITLE
Binding attached property on control code behind

### DISCUSSION THREAD

**Discussion by Mochengvia** [AvaloniaUI/Avalonia#11497] [General]
How do I bind a property on a control to another attached property? I checked all the docs and there is no mention of this.
```
//Xaml Code
<Setter Property="(i:VisualStateHelper.PressedBackground)"
    Value="{Binding Path=(i:VisualStateHelper.HoverBackground), RelativeSource={RelativeSource Self}}"/>
```

I've tried (the code is written in VisualStateHelper.cs):
```
control.Bind(PressedBackgroundProperty, new Binding()
{
    Path = "HoverBackground",
    Source = control,
});
```

```
control.Bind(PressedBackgroundProperty, new Binding()
{
    Path = "Panuon.Avalonia.UI.Internal.VisualStateHelper.HoverBackground",
    Source = control,
});
```

None of these worked.

**Comment by maxkatz6** [comment]
If you need it OneWay, this is way simpler solution:
`control.GetObservable(VisualStateHelper.HoverBackground).ToBinding()`

**Comment by Mochengvia** [comment]
It works, thanks :)
```
control.Bind(PressedBackgroundProperty,
    control.GetObservable(HoverBackgroundProperty).ToBinding());
```

**Comment by Mochengvia** [comment]
It seems that the binding created in this way has `Local` priority, how can I create a binding with other priority?

**Reply by maxkatz6** [reply]
You can specify priority in Bind method

**Comment by Mochengvia** [comment]
I try to use this code
```
control.Bind(HoverBackgroundProperty,
                    control.GetObservable(TemplatedControl.BackgroundProperty).ToBinding(),
                    BindingPriority.TemplatedParent);
```
to replace this
```
control.Bind(HoverBackgroundProperty, new Binding()
{
    Path = "Background",
    Source = control,
    Mode = BindingMode.OneWay,
    Priority = BindingPriority.TemplatedParent
});
```
But I can no longer modify the value of `HoverBackground` from xaml. Any ideas? 
It also doesn't work if I bind from a custom `OneWay` attached property to another `OneWay` attached property.
```
control.Bind(PressedBackgroundProperty,
    control.GetObservable(HoverBackgroundProperty).ToBinding(),
    BindingPriority.TemplatedParent);
```
Once the binding is set like this, I cannot change it's value from xaml.
